src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -218,7 +218,117 @@ private static unsafe int IndexOfVectorized(ref byte searchSpace, byte value, in
         exitFixed:;
                 return offset;
             }
+        }
+
+        public static bool TryIndicesOf(this Span<byte> buffer, byte value, Span<int> indices, out int numberOfIndices)
+        {
+            var length = buffer.Length;
+            if (length == 0 || indices.Length == 0)
+            {
+                numberOfIndices = 0;
+                return false;
+            }
+
+            return TryIndicesOf(ref buffer.DangerousGetPinnableReference(), value, length, indices, out numberOfIndices);
+        }
+
+        public static bool TryIndicesOf(this ReadOnlySpan<byte> buffer, byte value, Span<int> indices, out int numberOfIndices)
```

Comment:
It seems that to be able to parse HTTP, we will need:
1. The ability to match Span<byte> values
2. The ability to terminate the search when some terminator found. 

Is that an API that we can build on top of this one?